bundles/mapping/layerswipe/publisher/SwipeTool.js

```diff
@@ -0,0 +1,71 @@
+import { AbstractPublisherTool } from '../../../framework/publisher2/tools/AbstractPublisherTool';
+import { SwipeToolhandler } from '../handler/SwipeToolHandler';
+import { SwipeToolComponent } from './SwipeToolComponent';
+
+export const SWIPE_ID = 'Oskari.mapframework.bundle.layerswipe.plugin.LayerSwipePlugin';
+
+class SwipeTool extends AbstractPublisherTool {
+    constructor (...args) {
+        super(...args);
+        this.index = 5;
+        this.group = 'additional';
+        this.handler = new SwipeToolhandler(this);
+    }
+
+    getTool () {
+        return {
+            id: SWIPE_ID,
+            title: Oskari.getMsg('LayerSwipe', 'tool.label'),
+            config: this.state.pluginConfig || {}
+        };
+    }
+
+    getComponent () {
+        return {
+            component: SwipeToolComponent,
+            handler: this.handler
+        };
+    }
+
+    init (data) {
+        const configuration = data?.configuration?.layerswipe || {};
+        // restore state to handler -> passing init data to it
+        this.handler.init(configuration);
+        if (configuration.conf) {
+            this.storePluginConf(configuration.conf);
+            this.setEnabled(true);
+            this.handler.syncState();
```

Comment:
Check if we can remove syncState() call here.

Reply:
Checked. Can't. Bummer.